java/vector/src/main/java/org/apache/arrow/vector/dictionary/DictionaryEncoder.java

```diff
@@ -31,29 +32,37 @@
  */
 public class DictionaryEncoder {
 
-  // TODO recursively examine fields?
+  private final DictionaryHashTable hashTable;
+  private final Dictionary dictionary;
+  private final BufferAllocator allocator;
+
+  //TODO pass custom hasher.
+
+  /**
+   * Construct an instance.
+   */
+  public DictionaryEncoder(Dictionary dictionary, BufferAllocator allocator) {
+    this.dictionary = dictionary;
+    this.allocator = allocator;
+    hashTable = new DictionaryHashTable(dictionary.getVector());
+    hashTable.ensureBuildTable();
+  }
 
   /**
    * Dictionary encodes a vector with a provided dictionary. The dictionary must contain all values in the vector.
    *
-   * @param vector     vector to encode
-   * @param dictionary dictionary used for encoding
+   * @param vector vector to encode
    * @return dictionary encoded vector
    */
-  public static ValueVector encode(ValueVector vector, Dictionary dictionary) {
```

Comment:
this is a useful utility method, I don't think it should be removed, it can make use of the the instantiated object under the hood.

Reply:
Right, thanks for pointing out, added back now. In this way we could make optimization in later versions meanwhile not breaking previous API.